tests/integration/test_charm_with_profile.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+import glob
+import time
+from pathlib import Path
+
+import lightkube
+import pytest
+import yaml
+from lightkube import codecs
+from lightkube.generic_resource import create_global_resource
+from pytest_operator.plugin import OpsTest
+
+basedir = Path("./").absolute()
+PROFILE_NAMESPACE = "profile-example"
+PROFILE_FILE_PATH = f"{basedir}/tests/integration/profile.yaml"
+PROFILE_FILE = yaml.safe_load(Path(PROFILE_FILE_PATH).read_text())
+KUBEFLOW_USER_NAME = PROFILE_FILE["spec"]["owner"]["name"]
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = "training-operator"
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    """Build the charm and deploy."""
+    charm_under_test = await ops_test.build_charm(".")
+    image_path = METADATA["resources"]["training-operator-image"]["upstream-source"]
+    resources = {"training-operator-image": image_path}
+
+    await ops_test.model.deploy(
+        charm_under_test, resources=resources, application_name=APP_NAME, trust=True
+    )
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=60 * 10
+    )
+
+    # Deploy kubeflow-roles and kubeflow-profiles to create a Profile
+    await ops_test.model.deploy(
+        entity_url="kubeflow-roles",
+        channel="latest/edge",
+        trust=True,
+    )
+    await ops_test.model.deploy(
+        entity_url="kubeflow-profiles",
+        channel="latest/edge",
+        trust=True,
+    )
+
+    await ops_test.model.wait_for_idle(
+        status="active", raise_on_blocked=True, raise_on_error=True, timeout=60 * 10
+    )
+
+
+@pytest.mark.parametrize("example", glob.glob("examples/*.yaml"))
+@pytest.mark.abort_on_fail
+async def test_authorization_for_creating_resources(
+    example, ops_test: OpsTest, lightkube_client, apply_profile
+):
+    """Assert a *Job can be created by a user in the user namespace."""
+    # Set up for creating an object of kind *Job
+    job_yaml = yaml.safe_load(Path(example).read_text())
+    training_job = job_yaml["kind"]
+    _, stdout, __ = await ops_test.run(
```

Comment:
Is this a mistype or is it intentional?
```suggestion
    _, stdout, _ = await ops_test.run(
```

Reply:
Why though, given that we don't use them anyway.

Reply:
My understanding is that is the convention, but if I'm wrong I can change it.

Reply:
Accepted the suggestion, though, no need to have a distinction.